### PR TITLE
🐛Remove invalid ClusterRoleBinding from CertManager

### LINKED ISSUE #2928: clusterctl: filter out cert-manager-leaderelection ClusterRoleBinding
**What steps did you take and what happened:**
1. Create a management cluster with clusterctl
1. Try to back it up using Velero
1. Backup fails because it can't find cert-manager-leaderelection (Velero tries to back up all referenced roles)

**What did you expect to happen:**
1. No failure

**Anything else you would like to add:**
https://github.com/jetstack/cert-manager/pull/2207 removed the cert-manager-leaderelection ClusterRoleBinding (there is another ClusterRoleBinding with a slightly different name that fulfills the same intent and actually works).

We should probably add code to the cert manager part of clusterctl that just filters out this ClusterRoleBinding by name when converting the bindata assets back to objects.

/kind bug
/help

### PR DESCRIPTION
**Pull request by wfernandes**
Removes the cert-manager-leaderelection ClusterRoleBinding from
cert-manager v0.11.0 manifest

**What this PR does / why we need it**:
To get velero backup working we needed to get rid of dangling ClusterRoleBinding `cert-manager-leaderelection` from cert-manager v0.11.0. This PR stores a local copy of the cert-manager v0.11.0 release manifest which has been edited with the CRB removed and bindata regenerated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2928 

**Release Notes**
- ⚠️ Removes the `cert-manager-leaderelection` ClusterRoleBinding from being installed during `clusterctl init`. For existing clusters, the `cert-manager-leaderelection` ClusterRoleBinding needs to be manually removed if running into issues regarding https://github.com/jetstack/cert-manager/pull/2207

### CODE REVIEW

**Comment by wfernandes** [comment]
Not sure where to keep the cert-manager yaml. It could also go into the top level `hack` directory. But feedback appreciated.

File: Makefile [hunk 1]
```diff
@@ -254,8 +254,8 @@ generate-bindata: $(KUSTOMIZE) $(GOBINDATA) clean-bindata $(CLOUDINIT_GENERATED)
 	# Package manifest YAML into a single file.
 	mkdir -p $(GOBINDATA_CLUSTERCTL_DIR)/manifest/
 	$(KUSTOMIZE) build $(GOBINDATA_CLUSTERCTL_DIR)/crd > $(GOBINDATA_CLUSTERCTL_DIR)/manifest/clusterctl-api.yaml
-	# Fetch the cert-manager manifest
-	curl -sL https://github.com/jetstack/cert-manager/releases/download/v0.11.0/cert-manager.yaml > "$(GOBINDATA_CLUSTERCTL_DIR)/manifest/${CERTMANAGER_COMPONENTS_GENERATED_FILE}"
+	# Copy the "fixed" cert-manager v0.11.0 manifest
```
Comment (fabriziopandini) [inline Makefile:257]: Is it possible to keep the manifest under `cmd/clusterctl/config` eventually in a subfolder, and thus avoid the copy step?

**Comment by voor** [comment]
This PR will be a great first step to then adding PSP for cert-manager, once this is done you'd just need to re-generate this yaml with the PSP stuff turned on.

**Comment by voor** [comment]
Related Issue (this would make doing this issue for cert-manager much easier): https://github.com/kubernetes-sigs/cluster-api/issues/2934

File: Makefile (hunk 1 above)
Comment (wfernandes) [reply Makefile:257]: Sure. How about in a dir like `cmd/clusterctl/config/assets` and I can modify the gobindata cmd to include that directory. WDYT?

**Comment by wfernandes** [comment]
@fabriziopandini I moved the cert-manager manifest under `cmd/clusterctl/config/assets`. Let me know if that works. Thanks.

**Comment by fabriziopandini** [comment]
/assign @vincepri 

@wfernandes, one additional question. what is the plan for existing clusters? to provide instruction to the users about how to manually delete the invalid role binding or are we planning to make clusterctl to delete it automatically?

**Comment by wfernandes** [comment]
That's a good point. We could add some documentation about it. I can make a note of deleting the ClusterRoleBinding in the Release Notes. 
I'm leaning more towards the documentation solution for now because we have #2635 that could revamp how we manage the cert-manager.

**Comment by wfernandes** [comment]
Also...`clusterctl delete` doesn't remove the cert-manager components so it doesn't matter I guess 🤷‍♂️

**Comment by ncdc** [comment]
+1 to relnotes/docs for removing the bogus binding from existing clusters.

**Comment by fabriziopandini** [comment]
> Also...clusterctl delete doesn't remove the cert-manager components so it doesn't matter I guess

This should be tracked by an issue

**Comment by wfernandes** [comment]
@ncdc I added my release notes in the description of my PR. Let me know if I should make any changes to that.

**Pull request merged**